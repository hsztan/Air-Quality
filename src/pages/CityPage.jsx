import { useSelector } from 'react-redux';
import City from '../components/City/City';

const CityPage = () => {
  const city = useSelector((state) => state.city);
  console.log(city);
  return <City city={city} />;
};

export default CityPage;
