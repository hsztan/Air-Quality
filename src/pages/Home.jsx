import { useParams } from 'react-router-dom';
import NavBar from '../components/NavBar/NavBar';
import Container from '../components/Container/Container';
import { continentsNames } from '../data/data';

const Home = () => {
  const { continent, country } = useParams();
  const continentName = continent ? continentsNames[continent] : null;

  return (
    <div className="container">
      <NavBar title={country || continentName || 'continents'} />
      <Container continent={continentName} country={country} />
    </div>
  );
};

export default Home;
