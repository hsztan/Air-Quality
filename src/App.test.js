import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from './redux/configureStore';

import CityPage from './pages/CityPage';
import Home from './pages/Home';

describe('Pages renders correctly', () => {
  it('Home page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('CityPage page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <CityPage />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    render(
      <Provider store={store}>
        <BrowserRouter>
          <CityPage />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
