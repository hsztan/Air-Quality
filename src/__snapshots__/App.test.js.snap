// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages renders correctly CityPage page renders correctly 1`] = `
Array [
  <nav
    className="nav-bar"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      &lt;
    </button>
    <h1 />
    <div
      className="actions"
    >
      <ul>
        <li>
          <input
            id="nav-search"
            onKeyDown={[Function]}
            placeholder="search"
            type="text"
          />
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="city"
  >
    <h1>
      City
    </h1>
    <p />
    <h2>
      Population
    </h2>
    <img
      alt="group of people"
      src="population.png"
    />
    <p />
    <h2>
      Air Quality
    </h2>
    <p />
  </div>,
]
`;

exports[`Pages renders correctly Home page renders correctly 1`] = `
<div
  className="container"
>
  <nav
    className="nav-bar"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      &lt;
    </button>
    <h1>
      continents
    </h1>
    <div
      className="actions"
    >
      <ul>
        <li>
          <input
            id="nav-search"
            onKeyDown={[Function]}
            placeholder="search"
            type="text"
          />
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="continents"
  >
    <a
      href="/AF"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(africa.png)",
        }
      }
    >
      <p>
        Africa
      </p>
    </a>
    <a
      href="/AS"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(asia.png)",
        }
      }
    >
      <p>
        Asia
      </p>
    </a>
    <a
      href="/EU"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(europe.png)",
        }
      }
    >
      <p>
        Europe
      </p>
    </a>
    <a
      href="/NA"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(north-america.webp)",
        }
      }
    >
      <p>
        North America
      </p>
    </a>
    <a
      href="/OC"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(oceania.png)",
        }
      }
    >
      <p>
        Oceania
      </p>
    </a>
    <a
      href="/SA"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(south-america.webp)",
        }
      }
    >
      <p>
        South America
      </p>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
       
    </a>
    <a
      href="/AN"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(antartica.png)",
        }
      }
    >
      <p>
        Antartica
      </p>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
       
    </a>
  </div>
</div>
`;
